overview/content-types/example.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, config, ... }:
+let
+  inherit (lib)
+    mkOption
+    types
+    optionalString
+    any
+    attrValues
+    ;
+
+  types' = import ../../projects/types.nix { inherit lib; };
+in
+{
+  options = {
+    inherit (types'.example.getSubOptions { }) description module tests;
+    example-snippet = mkOption {
+      type = types.submodule ./code-snippet.nix;
+      default.filepath = config.module;
+    };
+    button-missing-test = mkOption {
+      type = types.str;
+      default = optionalString (any (test: test.module == null) (attrValues config.tests)) ''
+        <button class="button example">
+        <a class = "heading" href="https://github.com/ngi-nix/ngipkgs/blob/main/CONTRIBUTING.md">Add example test</a>
+        </button>
+      '';
+    };
```

Comment:
Please put this in a `let` block inside the `__toString` implementation. See the other PR by @eljamm which had the same pattern.